src/tsolvers/egraph/UFInterpolator.cc

```diff
@@ -348,10 +348,15 @@ bool UFInterpolator::colorEdgesFrom(CNode * x) {
 // formula into A and B.
 //
 PTRef
-UFInterpolator::getInterpolant(const ipartitions_t &, std::map<PTRef, icolor_t> * labels, PartitionManager & ) {
+UFInterpolator::getInterpolant(const ipartitions_t &, std::map<PTRef, icolor_t> * labels, PartitionManager &) {
     assert(labels);
+    if (labels) {
+        colorInfo.reset(new LocalTermColorInfo(*labels, logic));
+        litColors = *labels;
+    } else {
+        throw OsmtInternalException("Error in UFInterpolator::getInterpolant! No labels passed");
+    }
     srand(2);
```

Comment:
This seeding of the random number generator here seems strange.  Maybe something to consider for future.

Reply:
I agree, but it has been there before. Should I remove it?